projects/libs/core-library-typescript/src/common-library/block-utils.ts

```diff
@@ -0,0 +1,38 @@
+
+import { types, logger } from ".";
+import { createHash } from "crypto";
+
+export namespace BlockUtils {
+  export function calculateBlockHash(blockComponents: {header: types.BlockHeader, body: types.BlockBody}): Buffer {
+    const hash = createHash("sha256");
+
+    hash.update(JSON.stringify(blockComponents.header));
+
+    hash.update(JSON.stringify(blockComponents.body));
+
+    return hash.digest();
+  }
+
+  export function buildBlock(blockComponents: {header: types.BlockHeader, body: types.BlockBody}): types.Block {
+    const blockHash = calculateBlockHash(blockComponents);
+
+    const block: types.Block = {
+      header: blockComponents.header,
+      body: blockComponents.body,
+      hash: blockHash
+    };
+
+    return block;
+  }
+
+  export function buildNextBlock(body: types.BlockBody, prevBlock?: types.Block): types.Block {
+    return buildBlock({
+      header: {
+        version: 0,
+        prevBlockHash: prevBlock ? prevBlock.hash : new Buffer(""),
+        height: prevBlock ? prevBlock.header.height + 1 : 0
```

Comment:
This should error. It shouldn't be possible to build a block with `height` 0 - this will always be the Genesis block, so if `prevBlock` is undefined - this method should error.

Reply:
It was added for testing purposes